src/Dfe.PlanTech.Domain/Questionnaire/Models/Section.cs

```diff
@@ -21,23 +21,18 @@ public class Section : ContentComponent, ISection
 
     public RecommendationPage[] Recommendations { get; init; } = Array.Empty<RecommendationPage>();
 
-    public RecommendationPage GetRecommendationForMaturity(Maturity maturity)
-     => Recommendations.FirstOrDefault(recommendation => recommendation.Maturity == maturity) ??
-        throw new KeyNotFoundException($"Could not find recommendation with maturity {maturity}");
+    public RecommendationPage? TryGetRecommendationForMaturity(Maturity maturity) => Recommendations.FirstOrDefault(recommendation => recommendation.Maturity == maturity);
 
-    public RecommendationPage GetRecommendationForMaturity(string maturity)
+    public RecommendationPage? GetRecommendationForMaturity(string? maturity)
     {
         if (maturity is null)
             maturity = string.Empty;
 
-        if (Enum.TryParse(maturity, out Maturity maturityResponse))
-        {
-            return GetRecommendationForMaturity(maturityResponse);
-        }
-        else
-        {
-            return GetRecommendationForMaturity(Maturity.Unknown);
-        }
+        Maturity maturityResponse;
+
+        if (!Enum.TryParse(maturity, out maturityResponse)) maturityResponse = Maturity.Unknown;
```

Comment:
Same here maybe? Or combine the two in some way, where if `maturity is null OR doesn't convert to the enum then return null`

Reply:
Yep, should be nice performance-wise for that: [641b2c2](https://github.com/DFE-Digital/plan-technology-for-your-school/pull/230/commits/641b2c2cbb8bf6f6af5f323e54471d2d99054355)